File: MascotasUN/src/main/java/gestorAplicacion/gestionPersonas/Perro.java
/*Clase que crea y gestiona objetos de tipo Perro*/

package gestorAplicacion.gestionPersonas;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.Map;

import gestorAplicacion.gestionVentas.Producto;
import gestorAplicacion.gestionVentas.Tienda;

public class Perro implements Mascota, Serializable {

	// Serializador Perro
	private static final long serialVersionUID = 1L;
    private static ArrayList<Perro> perros = new ArrayList<>();
    
    // Atributos
    private long mascotaID;
    private String nombre;
    private int edad;
    private String raza;
    
    // Constructor
    public Perro(long mascotaID, String nombre, int edad, String raza) {
        this.mascotaID = mascotaID;
    	this.nombre = nombre;
        this.edad = edad;
        this.raza = raza;
        perros.add(this);
    }

    // Métodos
    
    // Método que dependiendo de la edad del perro (ingresada en años) retornará si es Cachorro o Adulto
    @Override
    public String calcularEdad() {
    	if (edad <= 1) {
    		return "Cachorro";
    	}
    	
    	else {
    		return "Adulto";
    	}
    }
    
    // Método que dependiendo de la edad y la raza del perro, retornará una lista de productos recomendados
    @Override
    public ArrayList<Producto> tipoAlimento(){
    	ArrayList<Producto> ListAux = new ArrayList<Producto>();
		for (int i = 0; i < Tienda.getCatalogo().size(); i++) {
			Producto p = Tienda.getCatalogo().get(i);
			if (p.getEtapa() == calcularEdad() & p.getRaza() == raza & p.getEspecie() == "Perro") {
				ListAux.add(p);
			}
		}
		return ListAux;
     }
    
    // Getters y setters
    
    public static ArrayList<Perro> getPerros() {
        return perros;
    }
    
    public static void setPerros(ArrayList<Perro> perros) {
        Perro.perros=perros;
    }
    
    public long getMascotaID() {
        return mascotaID;
    }
    
    public void setMascotaID(long mascotaID) {
        this.mascotaID = mascotaID;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public int getEdad() {
        return edad;
    }

    public void setEdad(int edad) {
        this.edad = edad;
    }

    public String getRaza() {
        return raza;
    }

    public void setRaza(String raza) {
        this.raza = raza;
    }
}

File: Practica2-MascotasUN/src/gestorAplicacion/gestionPersonas/Perro.py

from gestionVentas.Tienda import Producto
from gestionVentas.Tienda import Tienda
from Mascota import Mascota

class Perro(Mascota):

    #Constructor
    def __init__(self, mascotaID, nombre, edad, raza):
        self._mascotaID = mascotaID
        self._nombre = nombre
        self._edad = edad
        self._raza = raza

    #Metodos

    # Metodo que dependiendo de la edad del gato (ingresada en anos) retornara si es Cachorro o Adulto
    def calcularEdad(self):
        if (self._edad <= 1):
            return "Cachorro"
        else:
            return "Adulto"

    # Metodo que dependiendo de la edad del gato, retornara una lista de productos recomendados
    def tipoAlimento(self):
        LAux = []
        for i in Tienda.getCatalogo:
            if (i isinstance Producto): 

    #Getters & Setters
    def getMascotaID(self):
        return self._mascotaID
    
    def setMascotaID(self, mascotaID):
        self._mascotaID = mascotaID

    def getNombre(self):
        return self._nombre

    def setNombre(self, nombre):
        self._nombre = nombre

    def getEdad(self):
        return self._edad

    def setEdad(self, edad):
        self._edad = edad

    def getRaza(self):
        return self._raza

    def setRaza(self, raza):
        self._raza = raza